src/Form/OgMembershipTypeForm.php

```diff
@@ -0,0 +1,126 @@
+<?php
+
+namespace Drupal\og\Form;
+
+use Drupal\Core\Entity\BundleEntityFormBase;
+use Drupal\Core\Entity\EntityManagerInterface;
+use Drupal\Core\Entity\EntityTypeInterface;
+use Drupal\Core\Form\FormStateInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Form handler for OG membership type forms.
+ */
+class OgMembershipTypeForm extends BundleEntityFormBase {
+
+  /**
+   * The entity manager.
+   *
+   * @var \Drupal\Core\Entity\EntityManagerInterface
+   */
+  protected $entityManager;
+
+  /**
+   * Constructs the OgMembershipTypeForm object.
+   *
+   * @param \Drupal\Core\Entity\EntityManagerInterface $entity_manager
+   *   The entity manager.
+   */
+  public function __construct(EntityManagerInterface $entity_manager) {
+    $this->entityManager = $entity_manager;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('entity.manager')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function form(array $form, FormStateInterface $form_state) {
+    $form = parent::form($form, $form_state);
+
+    $type = $this->entity;
+    if ($this->operation == 'add') {
+      $form['#title'] = $this->t('Add membership type');
+    }
+    else {
+      $form['#title'] = $this->t('Edit %label membership type', ['%label' => $type->label()]);
+    }
+
+    $form['name'] = [
+      '#title' => $this->t('Name'),
+      '#type' => 'textfield',
+      '#default_value' => $type->label(),
+      '#description' => $this->t('The human-readable name of this membership type.'),
+      '#required' => TRUE,
+      '#size' => 30,
+    ];
+
+    $form['type'] = [
+      '#type' => 'machine_name',
+      '#default_value' => $type->id(),
+      '#maxlength' => EntityTypeInterface::BUNDLE_MAX_LENGTH,
+      '#machine_name' => [
+        'exists' => ['Drupal\og\Entity\OgMembershipType', 'load'],
+        'source' => ['name'],
+      ],
+      '#description' => $this->t('A unique machine-readable name for this membership type. It must only contain lowercase letters, numbers, and underscores.'),
+    ];
+    return $this->protectBundleIdElement($form);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function actions(array $form, FormStateInterface $form_state) {
+    $actions = parent::actions($form, $form_state);
+    $actions['submit']['#value'] = $this->t('Save membership type');
+    $actions['delete']['#value'] = $this->t('Delete membership type');
+    return $actions;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function validateForm(array &$form, FormStateInterface $form_state) {
```

Comment:
@jludwig 
Potentially here we could apply stricter rules to the names to avoid problems with crazy names like `<script>alert("xss");</script>` as described at https://github.com/Gizra/og/pull/231#issuecomment-282235558